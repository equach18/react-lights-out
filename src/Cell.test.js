import React from "react";
import Cell from "./Cell";
import { render, cleanup } from "@testing-library/react";

describe("Component: <Cell />", () => {
  let container;
  beforeEach(() => {
    const table = document.createElement("table");
    const tbody = document.createElement("tbody");
    const tr = document.createElement("tr");
    table.appendChild(tbody);
    tbody.appendChild(tr);
    container = tr;
    document.body.appendChild(table);
  });
  afterEach(() => {
    cleanup();
    container = null;
  });
  it("smoke test: renders without crashing", () => {
    render(<Cell />, { container });
  });

  it("matches snapshot: isLit=false", () => {
    const { asFragment } = render(<Cell isLit={false} />, { container });
    expect(asFragment()).toMatchSnapshot();
  });
  it("matches snapshot: isLit=true", () => {
    const { asFragment } = render(<Cell isLit />, { container });
    expect(asFragment()).toMatchSnapshot();
  });
});
