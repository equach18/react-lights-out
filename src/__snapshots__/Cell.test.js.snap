// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Cell /> matches snapshot: isLit=false 1`] = `<DocumentFragment />`;

exports[`Component: <Cell /> matches snapshot: isLit=true 1`] = `<DocumentFragment />`;
